### PR TITLE
Upgrade dask

### PR DESCRIPTION
**Pull request by mcgibbon**
The dask version we currently use has a security vulnerability. This PR upgrades that version.

### CODE REVIEW

File: fv3gfs-util/requirements.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ black==19.10b0
 numcodecs==0.7.2 #pin for gt4py, h5py and py3.6 to agree
 h5py==2.10.0 #pin for gt4py, h5py and py3.6 to agree
 h5netcdf
-dask==2021.03.0 #py3.6 - last supported version
+dask>=2021.10.0
```
Comment (elynnwu) [inline fv3gfs-util/requirements.txt:18]: You need to also update the constraints at the top level

**Review by elynnwu — APPROVED** [review]

File: fv3gfs-util/requirements.txt (hunk 1 above)
Comment (mcgibbon) [reply fv3gfs-util/requirements.txt:18]: Updated.

File: fv3core/requirements/requirements_base.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 f90nml
-pytest-xdist
+pytest-xdist<2
```
Comment (mcgibbon) [inline fv3core/requirements/requirements_base.txt:2]:
```suggestion
pytest-xdist<2  # last version that works with pytest 5
```

File: fv3core/requirements/requirements_base.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 f90nml
-pytest-xdist
+pytest-xdist<2  # last version that works with pytest 5
```
Comment (mcgibbon) [inline fv3core/requirements/requirements_base.txt:2]: pytest 5 is required by gt4py. It looks like something changed and now gt4py requirements are picked up when compiling constraints.txt.

**Comment by mcgibbon** [comment]
Closing, dask was updated in #64.

**Pull request closed without merging**